app/services/remove_provider_user.rb

```diff
@@ -0,0 +1,16 @@
+class RemoveProviderUser
+  attr_reader :current_provider_user, :user_to_remove
+
+  def initialize(current_provider_user:, user_to_remove:)
+    @current_provider_user = current_provider_user
+    @user_to_remove = user_to_remove
+  end
+
+  # Removes associations to providers common to the managing user
+  # and the user we're editing/removing.
+  def call!
+    shared_providers = current_provider_user.providers & user_to_remove.providers
```

Comment:
Should this be `current_provider_user.providers.manage_users`?

Reply:
https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training/pull/2012/commits/e659aabb6938ba7684f75fd56bd065e7ab80b07e#diff-1a42958a6311b2501e78d6b964484de7R12 scopes this to managed providers.
I wonder about the use of `#map` here though, I think we should be able to achieve this with a join. I know we saw some non-uniq results in the past using a join but I wonder if this was the result of a bug which caused duplicate `ProviderPermission` records which has been fixed by https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training/pull/1931.